src/activity_actor.cpp

```diff
@@ -2532,6 +2538,466 @@ std::unique_ptr<activity_actor> milk_activity_actor::deserialize( JsonIn &jsin )
     return actor.clone();
 }
 
+void read_activity_actor::start( player_activity &act, Character &/*who*/ )
+{
+    if( !book->is_book() ) {
+        debugmsg( "ACT_READ on a non-book item" );
+        act.set_to_null();
+        return;
+    }
+
+    if( book->type->use_methods.count( "MA_MANUAL" ) ) {
+        bktype = static_cast<int>( book_type::martial_art );
+    }
```

Comment:
Instead it can always be retrieved from the book type.

Reply:
I use it mainly for the code in `read_activity_actor::do_turn`
```c++
if( bktype == static_cast<int>( book_type::martial_art ) && one_in( 3 ) ) {
    who.mod_stamina( -1 );
}
```
that way I won't need to call `book->type->use_methods.count( "MA_MANUAL" )` every turn and only once at the start. I could check if its a martial arts book when deserializing but I would like to keep that function simple like the other deserialization functions. I also used a `enum` for book type instead of a `bool` to accommodate future different book types.

Reply:
You can instead cache the value when `do_turn` is called (use `cata::optional`).

```
cata::optional<book_type> bktype;
```

```
if( !bktype.has_value() ) {
    bktype = book->type->use_methods.count( "MA_MANUAL" );
}
```

Reply:
Thanks. I didn't know how to cache.